src/PopChainManager/index.ts

```diff
@@ -8,6 +8,35 @@ export class PopChainManager {
     return closest(pop.targetEl, '[pop]', true);
   }
 
+  public getFullPopChain(pop: Pop, hideFullChain: boolean): Pop[] {
```

Comment:
```
 +  public getFullPopChain(pop: Pop, hideFullChain: boolean): Pop[] {
 +    let popChain: Pop[] = [];
 +
 +    if (!!pop.childPops.length) {
 +      pop.childPops.forEach(function(child: Pop): void {
 +        let idx = child.parentPop.childPops.indexOf(child);
 +        if (idx !== -1) {
 +          child.parentPop.childPops.splice(idx, 1);
 +        }
 +        child.parentPop = null;
 +        popChain = popChain.concat(this.getFullPopChain(child, hideFullChain));
 +      }, this);
 +    }
 +
 +    popChain.push(pop);
 +
 +    if (!!pop.parentPop) {
 +      let idx = pop.parentPop.childPops.indexOf(pop);
 +      if (idx !== -1) {
 +        pop.parentPop.childPops.splice(idx, 1);
 +      }
 +      if (!pop.parentPop.isPinned || hideFullChain) {
 +        popChain = popChain.concat(this.getFullPopChain(pop.parentPop, hideFullChain));
 +      }
 +    }
 +
 +    return popChain;
 +  }
```

Is this the equivalent of the code below? Difference being that this approach is an iterative BFS and adds parent pop first? (This is untested, just wrote it during the Caltrain ride and seeing if I fully grasp the intent of this function)

```
   public getFullPopChain(pop:Pop, hideFullChain: boolean): Pop[] {
      let popChain: Pop[] = [];
      if (hideFullChain) {
         pop = getRootPop(pop);
      } else {
         pop = getFirstUnpinnedParentPop(pop);
      }

      let stack: Pop[] = [];
      stack.push(pop);

      while(stack.length) {
         pop = stack.pop();
         popChain.push(pop);

         if (!!pop.childPops.length) {
            pop.childPops.forEach(function(child: Pop, index: number): void {
               pop.childPops.splice(index, 1);
               /* Removes reference to parent since the parent now disowned the child. This is also a mutation that probably does not belong here, or rename method. */
               child.parentPop = null;
               stack.push(child);
            }
         }
      }
      return popChain;
   }

   public getRootPop(pop: Pop) {
      if (pop) {
         while (pop.parentPop) {
            pop = pop.parentPop;
         }
         return pop;
      }
   }

   public getFirstUnpinnedParentPop(pop: Pop) {
      if (pop) {
         pop = pop.parentPop;

         while (pop && pop.isPinned) {
            pop = pop.parentPop;
         }

         return pop;
   }
```